core/src/main/java/org/elasticsearch/http/netty/NettyHttpChannel.java

```diff
@@ -134,6 +137,7 @@ public void sendResponse(RestResponse response) {
             if (!addedReleaseListener && content instanceof Releasable) {
                 ((Releasable) content).close();
             }
+            transport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
```

Comment:
The reason why I suggested AutoClosable is that I don't want close to be idempotent, but rather throw an exception, see Closable docs:

```
  If the stream is already closed then invoking this method has no effect.
```

I'm fine with not implementing any interface at all and just implement the close() method and call it on sendResponse